packages/jest-cli/src/runJest.js

```diff
@@ -174,6 +174,14 @@ const runJest = async (
   );
 
   allTests = sequencer.sort(allTests);
+
+  // With --listTests, simply print the test info as JSON and exit.
+  if (globalConfig.listTests) {
+    const testsJson = JSON.stringify(allTests, null, '  ');
+    new Console(pipe, pipe).log(testsJson);
+    process.exit(0);
```

Comment:
I know I mentioned exiting earlier but I think it would be better if we could instead `return {success: true};` which will allow Jest to exit cleanly. Can you try to see if that works? :)